core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -296,6 +309,14 @@ protected Logger logger() {
         return DockerLoggerFactory.getLogger(this.getDockerImageName());
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Logger getLogger() {
```

Comment:
a leftover? Logger should not be exposed

Reply:
The `execInContainer()` methods, which have now been moved to `ExecInContainerPattern` still use a logger which must be passed into the method.

It's used for a debug and two trace statements, which are useful, but not exactly crucial.
Should they be removed, or can you see another way around this?

Reply:
Yes, I think we should remove it and create another logger in `ExecInContainerPattern`